### Contents

#### Import and check data for missing values

#### Check for duplicates and check data types

#### Basic statistical analysis and check for anomalous values

In [10]:
import pandas as pd
import numpy as np
import os

In [12]:
# import Toronto Fire Incidents data

tfire = pd.read_csv (r"C:\Users\agwsi\CF\6. Advanced Analytics and Dashboard design\Project folder\2. Orginal Data\Fire-Incidents.csv")

In [18]:
# explore basic properties

tfire.shape

(11214, 27)

In [22]:
# allow all columns to be shown

pd.set_option('display.max_columns', None)

In [24]:
tfire.head()

Area_of_Origin  \
0                                   Porch or Balcony   
1                            Cooking Area or Kitchen   
2     Living Area (e.g. living, TV, recreation, etc)   
3                        Undetermined  (formerly 98)   
4  Sleeping Area or Bedroom (inc. patients room, ...   

                   Business_Impact  Civilian_Casualties  \
0         No business interruption                    0   
1                     Undetermined                    0   
2  Not applicable (not a business)                    1   
3        May not resume operations                    0   
4  Not applicable (not a business)                    8   

   Count_of_Persons_Rescued  Estimated_Dollar_Loss  \
0                        86                   3000   
1                        28                  50000   
2                        16                1000000   
3                        12                1000000   
4                        11                 125000   

   Estimated_Number_Of_Persons_Displaced Ext_agent_app_or_defer_time  \
0                                      0         2018-08-24T17:06:26   
1                                     28         2018-11-24T07:19:00   
2                                    130         2017-02-09T18:02:13   
3                                     14         2012-10-30T00:52:04   
4                                      2         2018-07-08T04:35:00   

                             Extent_Of_Fire  \
0              Confined to object of origin   
1   Confined to part of room/area of origin   
2  Spread beyond room of origin, same floor   
3                          Entire Structure   
4  Spread beyond room of origin, same floor   

              Fire_Alarm_System_Impact_on_Evacuation  \
0  Not applicable: Occupant(s) first alerted by o...   
1  Not applicable: Occupant(s) first alerted by o...   
2  Some persons (at risk) evacuated as a result o...   
3                                       Undetermined   
4  Some persons (at risk) evacuated as a result o...   

                Fire_Alarm_System_Operation Fire_Alarm_System_Presence  \
0         Fire alarm system did not operate  Fire alarm system present   
1                Fire alarm system operated  Fire alarm system present   
2                Fire alarm system operated  Fire alarm system present   
3  Fire alarm system operation undetermined               Undetermined   
4                Fire alarm system operated  Fire alarm system present   

  Fire_Under_Control_Time                                    Ignition_Source  \
0     2018-08-24T17:06:28                                       Undetermined   
1     2018-11-24T07:19:00  Other Cooking Items (eg Toaster, Kettle, elec ...   
2     2017-02-09T18:36:21                                       Undetermined   
3     2012-10-30T01:28:48                                       Undetermined   
4     2018-07-08T04:38:28        Matches or Lighters (unable to distinguish)   

   Incident_Station_Area  Incident_Ward Last_TFS_Unit_Clear_Time   Latitude  \
0                    114           18.0      2018-08-24T17:42:43  43.760357   
1                    142            7.0      2018-11-24T12:45:21  43.754542   
2                    325           27.0      2017-02-09T23:36:33  43.657300   
3                    332           20.0      2012-11-01T16:30:31  43.649350   
4                    442            5.0      2018-07-08T16:44:07  43.697960   

   Longitude      Material_First_Ignited           Method_Of_Fire_Control  \
0 -79.412479  Undetermined (formerly 98)  Extinguished by fire department   
1 -79.530419                     Plastic         Extinguished by occupant   
2 -79.373460  Undetermined (formerly 98)  Extinguished by fire department   
3 -79.393700  Undetermined (formerly 98)  Extinguished by fire department   
4 -79.511539                     Bedding  Extinguished by fire department   

        Possible_Cause                         Property_Use  \
0         Undetermined  Multi-Unit Dwelli

In [30]:
# check for missing values

tfire.isnull().sum()

Area_of_Origin                               0
Business_Impact                              0
Civilian_Casualties                          0
Count_of_Persons_Rescued                     0
Estimated_Dollar_Loss                        0
Estimated_Number_Of_Persons_Displaced        0
Ext_agent_app_or_defer_time                  0
Extent_Of_Fire                               0
Fire_Alarm_System_Impact_on_Evacuation       0
Fire_Alarm_System_Operation                  0
Fire_Alarm_System_Presence                   0
Fire_Under_Control_Time                      0
Ignition_Source                              0
Incident_Station_Area                        0
Incident_Ward                               33
Last_TFS_Unit_Clear_Time                     0
Latitude                                     0
Longitude                                    0
Material_First_Ignited                       0
Method_Of_Fire_Control                       0
Possible_Cause                               0
Property_Use 

In [36]:
# investigate miisng values from 'Incident_Ward'

missing = tfire[tfire['Incident_Ward'].isnull()]

In [40]:
missing.head()

Area_of_Origin  \
907                                              Garage   
1442                                 Electrical Systems   
1921                Mechanical/Electrical Services Room   
2973  Sleeping Area or Bedroom (inc. patients room, ...   
3193                            Cooking Area or Kitchen   

                      Business_Impact  Civilian_Casualties  \
907                      Undetermined                    0   
1442         No business interruption                    0   
1921         No business interruption                    0   
2973  Not applicable (not a business)                    0   
3193  Not applicable (not a business)                    0   

      Count_of_Persons_Rescued  Estimated_Dollar_Loss  \
907                          0                      0   
1442                         0                   4000   
1921                         0                   1000   
2973                         0                  20000   
3193                         0                   5000   

      Estimated_Number_Of_Persons_Displaced Ext_agent_app_or_defer_time  \
907                                       0         2019-01-25T18:57:07   
1442                                      0         2019-02-15T12:20:10   
1921                                      0         2018-07-11T10:41:04   
2973                                      7         2011-02-12T18:15:00   
3193                                      6         2011-05-14T18:33:00   

                               Extent_Of_Fire  \
907              Confined to object of origin   
1442             Confined to object of origin   
1921  Confined to part of room/area of origin   
2973          Spread to entire room of origin   
3193  Confined to part of room/area of origin   

                 Fire_Alarm_System_Impact_on_Evacuation  \
907   Not applicable: No fire alarm system, no perso...   
1442  All persons (at risk of injury) evacuated as a...   
1921                                       Undetermined   
2973  Not applicable: No fire alarm system, no perso...   
3193  Not applicable: No fire alarm system, no perso...   

                   Fire_Alarm_System_Operation  \
907                 Fire alarm system operated   
1442                Fire alarm system operated   
1921  Fire alarm system operation undetermined   
2973                Not applicable (no system)   
3193                Not applicable (no system)   

                             Fire_Alarm_System_Presence  \
907                           Fire alarm system present   
1442                          Fire alarm system present   
1921                                       Undetermined   
2973  Not applicable (bldg not classified by OBC OR ...   
3193  Not applicable (bldg not classified by OBC OR ...   

     Fire_Under_Control_Time  \
907      2019-01-25T19:03:12   
1442     2019-02-15T12:16:19   
1921     2018-07-11T10:36:33   
2973     2011-02-12T18:50:50   
3193     2011-05-14T18:35:52   

                                        Ignition_Source  \
907                                 Under Investigation   
1442  Distribution Equipment (includes panel boards,...   
1921  Distribution Equipment (includes panel boards,...   
2973                                       Undetermined   
3193                            Stove, Range-top burner   

      Incident_Station_Area  Incident_Ward Last_TFS_Unit_Clear_Time  \
907                     141            NaN      2019-01-25T19:22:50   
1442                    141            NaN      2019-02-15T12:49:09   
1921                    411            NaN      2018-07-11T11:52:09   
2973                    242            NaN      2011-02-12T19:52:31   
3193                    322            NaN      2011-05-14T19:40:33   

       Latitude  Longitude        Material_First_Ignited  \
907   43.783218 -79.487306    Undetermined (formerly 98)   
1442  43.777718 -79.511093  Electrical Wiring Insulation   
1921  43.770920 -79.543448                    Insulation  

In [42]:
# a brief look shows that there are no civilian casualties and no persons rescued when the incident ward value is null.
# investigate if this is true for the entire dataset to explain missing values

missing_ward_with_casualties = tfire[(tfire['Incident_Ward'].isnull()) & (tfire['Civilian_Casualties'] != 0)]

In [62]:
missing_ward_with_casualties.head()

Area_of_Origin                  Business_Impact  \
4326                    Garage         No business interruption   
10757  Cooking Area or Kitchen  Not applicable (not a business)   
10919  Cooking Area or Kitchen  Not applicable (not a business)   

       Civilian_Casualties  Count_of_Persons_Rescued  Estimated_Dollar_Loss  \
4326                     1                         0                  10000   
10757                    1                         0                    500   
10919                    1                         0                   4000   

       Estimated_Number_Of_Persons_Displaced Ext_agent_app_or_defer_time  \
4326                                       0         2017-11-28T14:36:00   
10757                                      0         2017-12-01T17:50:59   
10919                                      0         2014-05-26T17:33:27   

                                Extent_Of_Fire  \
4326   Confined to part of room/area of origin   
10757  Confined to part of room/area of origin   
10919  Confined to part of room/area of origin   

                  Fire_Alarm_System_Impact_on_Evacuation  \
4326   Not applicable: No fire alarm system, no perso...   
10757  Not applicable: No fire alarm system, no perso...   
10919  No one (at risk) evacuated as a result of hear...   

      Fire_Alarm_System_Operation  \
4326   Not applicable (no system)   
10757  Not applicable (no system)   
10919  Fire alarm system operated   

                              Fire_Alarm_System_Presence  \
4326                                No Fire alarm system   
10757  Not applicable (bldg not classified by OBC OR ...   
10919                          Fire alarm system present   

      Fire_Under_Control_Time          Ignition_Source  Incident_Station_Area  \
4326      2017-11-28T14:36:00             Undetermined                    133   
10757     2017-12-01T17:50:59  Stove, Range-top burner                    113   
10919     2014-05-26T17:33:20  Stove, Range-top burner                    313   

       Incident_Ward Last_TFS_Unit_Clear_Time  Latitude  Longitude  \
4326             NaN      2017-11-28T21:59:41  43.70865  -79.46855   
10757            NaN      2017-12-01T18:27:39  43.79745  -79.33094   
10919            NaN      2014-05-26T18:08:20  43.66730  -79.37378   

      Material_First_Ignited     Method_Of_Fire_Control Possible_Cause  \
4326   Rubbish, Trash, Waste   Extinguished by occupant   Undetermined   
10757    Cooking Oil, Grease   Extinguished by occupant     Unattended   
10919    Cooking Oil, Grease  Action taken unclassified     Unattended   

                                            Property_Use  \
4326                 Other Business or Personal Services   
10757  Attached Dwelling (eg. rowhouse, townhouse, etc.)   
10919  Attached Dwelling (eg. rowhouse, townhouse, etc.)   

                Smoke_Alarm_at_Fire_Origin_Alarm_Failure  \
4326                                          No battery   
10757  Not applicable: Alarm operated OR presence/ope...   
10919  Not applicable: Alarm operated OR presence/ope...   

                   Smoke_Alarm_at_Fire_Origin_Alarm_Type  \
4326   Not applicable - no smoke alarm or presence un...   
10757                                     Interconnected   
10919                             Hardwired (standalone)   

                               Status_of_Fire_On_Arrival       TFS_Alarm_Time  \
4326                  Fire extinguished prior to arrival  2017-11-28T14:27:02   
10757                 Fire extinguished prior to arrival  2017-12-01T17:40:21   
10919  Fire with smoke showing only - including vehic...  2014-05-26T17:22:50   

          TFS_Arrival_Time  
4326   2017-11-28T14:33:01  
10757  2017-12-01T17:46:49  
10919  2014-05-26T17:26:59

In [46]:
# check for persons rescued column

missing_ward_with_rescues = tfire[(tfire['Incident_Ward'].isnull()) & (tfire['Count_of_Persons_Rescued'] != 0)]

In [60]:
missing_ward_with_rescues.head()

Empty DataFrame
Columns: [Area_of_Origin, Business_Impact, Civilian_Casualties, Count_of_Persons_Rescued, Estimated_Dollar_Loss, Estimated_Number_Of_Persons_Displaced, Ext_agent_app_or_defer_time, Extent_Of_Fire, Fire_Alarm_System_Impact_on_Evacuation, Fire_Alarm_System_Operation, Fire_Alarm_System_Presence, Fire_Under_Control_Time, Ignition_Source, Incident_Station_Area, Incident_Ward, Last_TFS_Unit_Clear_Time, Latitude, Longitude, Material_First_Ignited, Method_Of_Fire_Control, Possible_Cause, Property_Use, Smoke_Alarm_at_Fire_Origin_Alarm_Failure, Smoke_Alarm_at_Fire_Origin_Alarm_Type, Status_of_Fire_On_Arrival, TFS_Alarm_Time, TFS_Arrival_Time]
Index: []

In [54]:
missing_rescues_and_wards = tfire[(tfire['Incident_Ward']>=1) & (tfire['Count_of_Persons_Rescued'] ==0)]

In [58]:
missing_rescues_and_wards.head()

Area_of_Origin  \
327                      Trash, rubbish area (outside)   
328  Sleeping Area or Bedroom (inc. patients room, ...   
329                Mechanical/Electrical Services Room   
330                                             Office   
331                            Cooking Area or Kitchen   

                         Business_Impact  Civilian_Casualties  \
327             No business interruption                    0   
328             No business interruption                    0   
329  May resume operations within a week                    0   
330             No business interruption                    0   
331      Not applicable (not a business)                    0   

     Count_of_Persons_Rescued  Estimated_Dollar_Loss  \
327                         0                      0   
328                         0                   2000   
329                         0                 100000   
330                         0                   5000   
331                         0                    500   

     Estimated_Number_Of_Persons_Displaced Ext_agent_app_or_defer_time  \
327                                      0         2018-02-25T14:19:25   
328                                      0         2018-02-26T18:28:00   
329                                      0         2018-02-27T10:57:32   
330                                      0         2018-02-25T15:57:00   
331                                      0         2018-02-26T18:16:50   

                              Extent_Of_Fire  \
327             Confined to object of origin   
328  Confined to part of room/area of origin   
329  Confined to part of room/area of origin   
330             Confined to object of origin   
331             Confined to object of origin   

                Fire_Alarm_System_Impact_on_Evacuation  \
327                                       Undetermined   
328  Not applicable: No fire alarm system, no perso...   
329  Some persons (at risk) evacuated as a result o...   
330  All persons (at risk of injury) evacuated as a...   
331  All persons (at risk of injury) evacuated as a...   

    Fire_Alarm_System_Operation  \
327  Not applicable (no system)   
328  Not applicable (no system)   
329  Fire alarm system operated   
330  Fire alarm system operated   
331  Fire alarm system operated   

                            Fire_Alarm_System_Presence  \
327                                       Undetermined   
328  Not applicable (bldg not classified by OBC OR ...   
329                          Fire alarm system present   
330                          Fire alarm system present   
331                          Fire alarm system present   

    Fire_Under_Control_Time  \
327     2018-02-25T14:20:00   
328     2018-02-26T18:30:00   
329     2018-02-27T11:36:09   
330     2018-02-25T15:58:00   
331     2018-02-26T18:16:50   

                                       Ignition_Source  Incident_Station_Area  \
327                                       Undetermined                    133   
328          Incandescent Lamp - Light Bulb, Spotlight                    235   
329  Distribution Equipment (includes panel boards,...                    231   
330                            Other Heating Equipment                    332   
331                            Stove, Range-top burner                    325   

     Incident_Ward Last_TFS_Unit_Clear_Time   Latitude  Longitude  \
327            5.0      2018-02-25T15:07:42  43.708659 -79.478062   
328           19.0      2018-02-26T19:05:58  43.714811 -79.304107   
329           24.0      2018-02-27T13:51:29  43.748582 -79.222367   
330           10.0      2018-02-25T19:55:41  43.652152 -79.382336   
331           13.0      2018-02-26T18:38:22  43.656807 -79.362569   

           Material_First_Ignited           Method_Of_Fire_Control  \
327    Undetermined (formerly 98)  Extinguished by fire department   
328                    Insulation  Extinguished by fire department   
329  Electrical Wirin

##### No clear reason found as to why there are missing values for the 'Incident_Ward' column.
##### A new value of '0' will be created for the missing wards

In [66]:
# replace all missing values with '0'

tfire['Incident_Ward'].fillna(0, inplace= True)

C:\Users\agwsi\AppData\Local\Temp\ipykernel_28436\3995252573.py:3: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  tfire['Incident_Ward'].fillna(0, inplace= True)


In [68]:
# check results

tfire.isnull().sum()

Area_of_Origin                              0
Business_Impact                             0
Civilian_Casualties                         0
Count_of_Persons_Rescued                    0
Estimated_Dollar_Loss                       0
Estimated_Number_Of_Persons_Displaced       0
Ext_agent_app_or_defer_time                 0
Extent_Of_Fire                              0
Fire_Alarm_System_Impact_on_Evacuation      0
Fire_Alarm_System_Operation                 0
Fire_Alarm_System_Presence                  0
Fire_Under_Control_Time                     0
Ignition_Source                             0
Incident_Station_Area                       0
Incident_Ward                               0
Last_TFS_Unit_Clear_Time                    0
Latitude                                    0
Longitude                                   0
Material_First_Ignited                      0
Method_Of_Fire_Control                      0
Possible_Cause                              0
Property_Use                      

In [74]:
# check for duplicates

tfire.duplicated().sum()

0

##### no duplicates found

##### check data types

In [82]:
tfire.dtypes

Area_of_Origin                               object
Business_Impact                              object
Civilian_Casualties                           int64
Count_of_Persons_Rescued                      int64
Estimated_Dollar_Loss                         int64
Estimated_Number_Of_Persons_Displaced         int64
Ext_agent_app_or_defer_time                  object
Extent_Of_Fire                               object
Fire_Alarm_System_Impact_on_Evacuation       object
Fire_Alarm_System_Operation                  object
Fire_Alarm_System_Presence                   object
Fire_Under_Control_Time                      object
Ignition_Source                              object
Incident_Station_Area                         int64
Incident_Ward                               float64
Last_TFS_Unit_Clear_Time                     object
Latitude                                    float64
Longitude                                   float64
Material_First_Ignited                       object
Method_Of_Fi

##### times are all in the object format so shall be converted to date and time formats

##### incident ward should be int64

##### estimated dollar loss should be changed to float64

In [99]:
# first check time columns to see how they currently appear

tfire [['Ext_agent_app_or_defer_time','Fire_Under_Control_Time', 'Last_TFS_Unit_Clear_Time', 'TFS_Alarm_Time', 'TFS_Arrival_Time']].head()

Ext_agent_app_or_defer_time Fire_Under_Control_Time  \
0         2018-08-24 17:06:26     2018-08-24T17:06:28   
1         2018-11-24 07:19:00     2018-11-24T07:19:00   
2         2017-02-09 18:02:13     2017-02-09T18:36:21   
3         2012-10-30 00:52:04     2012-10-30T01:28:48   
4         2018-07-08 04:35:00     2018-07-08T04:38:28   

  Last_TFS_Unit_Clear_Time       TFS_Alarm_Time     TFS_Arrival_Time  
0      2018-08-24T17:42:43  2018-08-24T16:49:36  2018-08-24T16:54:09  
1      2018-11-24T12:45:21  2018-11-24T07:09:12  2018-11-24T07:14:23  
2      2017-02-09T23:36:33  2017-02-09T17:45:07  2017-02-09T17:48:49  
3      2012-11-01T16:30:31  2012-10-30T00:42:01  2012-10-30T00:44:58  
4      2018-07-08T16:44:07  2018-07-08T04:08:50  2018-07-08T04:13:54

##### all columns are using the same format and will be converted to datetime format

In [97]:
tfire['Ext_agent_app_or_defer_time'] = pd.to_datetime(tfire['Ext_agent_app_or_defer_time'])

In [101]:
tfire['Fire_Under_Control_Time'] = pd.to_datetime(tfire['Fire_Under_Control_Time'])

In [103]:
tfire['Last_TFS_Unit_Clear_Time'] =pd.to_datetime(tfire['Last_TFS_Unit_Clear_Time'])

In [105]:
tfire ['TFS_Alarm_Time'] = pd.to_datetime(tfire['TFS_Alarm_Time'])

In [107]:
tfire ['TFS_Arrival_Time'] = pd.to_datetime(tfire['TFS_Arrival_Time'])

In [109]:
# check results

tfire [['Ext_agent_app_or_defer_time','Fire_Under_Control_Time', 'Last_TFS_Unit_Clear_Time', 'TFS_Alarm_Time', 'TFS_Arrival_Time']].head()

Ext_agent_app_or_defer_time Fire_Under_Control_Time  \
0         2018-08-24 17:06:26     2018-08-24 17:06:28   
1         2018-11-24 07:19:00     2018-11-24 07:19:00   
2         2017-02-09 18:02:13     2017-02-09 18:36:21   
3         2012-10-30 00:52:04     2012-10-30 01:28:48   
4         2018-07-08 04:35:00     2018-07-08 04:38:28   

  Last_TFS_Unit_Clear_Time      TFS_Alarm_Time    TFS_Arrival_Time  
0      2018-08-24 17:42:43 2018-08-24 16:49:36 2018-08-24 16:54:09  
1      2018-11-24 12:45:21 2018-11-24 07:09:12 2018-11-24 07:14:23  
2      2017-02-09 23:36:33 2017-02-09 17:45:07 2017-02-09 17:48:49  
3      2012-11-01 16:30:31 2012-10-30 00:42:01 2012-10-30 00:44:58  
4      2018-07-08 16:44:07 2018-07-08 04:08:50 2018-07-08 04:13:54

In [114]:
tfire.dtypes

Area_of_Origin                                      object
Business_Impact                                     object
Civilian_Casualties                                  int64
Count_of_Persons_Rescued                             int64
Estimated_Dollar_Loss                                int64
Estimated_Number_Of_Persons_Displaced                int64
Ext_agent_app_or_defer_time                 datetime64[ns]
Extent_Of_Fire                                      object
Fire_Alarm_System_Impact_on_Evacuation              object
Fire_Alarm_System_Operation                         object
Fire_Alarm_System_Presence                          object
Fire_Under_Control_Time                     datetime64[ns]
Ignition_Source                                     object
Incident_Station_Area                                int64
Incident_Ward                                      float64
Last_TFS_Unit_Clear_Time                    datetime64[ns]
Latitude                                           float

In [119]:
# changing incident ward data type from float64 to int64

tfire['Incident_Ward'] = tfire['Incident_Ward'].astype('int64')

In [121]:
# changing estimated dollar loss to float64

tfire['Estimated_Dollar_Loss'] = tfire['Estimated_Dollar_Loss'].astype('float64')

In [123]:
#check results

tfire.dtypes

Area_of_Origin                                      object
Business_Impact                                     object
Civilian_Casualties                                  int64
Count_of_Persons_Rescued                             int64
Estimated_Dollar_Loss                              float64
Estimated_Number_Of_Persons_Displaced                int64
Ext_agent_app_or_defer_time                 datetime64[ns]
Extent_Of_Fire                                      object
Fire_Alarm_System_Impact_on_Evacuation              object
Fire_Alarm_System_Operation                         object
Fire_Alarm_System_Presence                          object
Fire_Under_Control_Time                     datetime64[ns]
Ignition_Source                                     object
Incident_Station_Area                                int64
Incident_Ward                                        int64
Last_TFS_Unit_Clear_Time                    datetime64[ns]
Latitude                                           float

##### Data types all properly corrected

##### Check statistics

In [128]:
tfire.describe()

Civilian_Casualties  Count_of_Persons_Rescued  Estimated_Dollar_Loss  \
count         11214.000000              11214.000000           1.121400e+04   
mean              0.112538                  0.062154           4.294369e+04   
min               0.000000                  0.000000           0.000000e+00   
25%               0.000000                  0.000000           2.500000e+02   
50%               0.000000                  0.000000           2.500000e+03   
75%               0.000000                  0.000000           1.500000e+04   
max              15.000000                 86.000000           5.000000e+07   
std               0.456429                  0.962587           5.339367e+05   

       Estimated_Number_Of_Persons_Displaced    Ext_agent_app_or_defer_time  \
count                           11214.000000                          11214   
mean                               17.274835  2015-03-24 00:25:33.527108864   
min                                 0.000000            2011-01-01 06:03:00   
25%                                 0.000000            2013-02-11 19:42:11   
50%                                 0.000000     2015-03-03 11:19:44.500000   
75%                                 1.000000            2017-05-14 06:16:45   
max                               999.000000            2019-06-30 22:58:00   
std                               119.071696                            NaN   

             Fire_Under_Control_Time  Incident_Station_Area  Incident_Ward  \
count                          11214           11214.000000   11214.000000   
mean   2015-03-24 00:34:04.513554432             288.281880      19.525058   
min              2011-01-01 06:13:00             111.000000       0.000000   
25%       2013-02-11 19:42:30.500000             221.000000       9.250000   
50%              2015-03-03 11:21:58             314.000000      19.000000   
75%    2017-05-14 06:17:44.750000128             346.000000      28.000000   
max              2019-06-30 22:59:00             445.000000      44.000000   
std                              NaN             107.515995      11.793265   

            Last_TFS_Unit_Clear_Time      Latitude     Longitude  \
count                          11214  11214.000000  11214.000000   
mean   2015-03-24 02:14:54.476101120     43.705190    -79.403323   
min              2011-01-01 07:29:21     43.588926    -79.635241   
25%       2013-02-11 20:19:46.500000     43.662474    -79.479273   
50%              2015-03-03 11:40:55     43.697050    -79.403892   
75%       2017-05-14 07:04:51.500000     43.747770    -79.337150   
max              2019-06-30 23:25:25     43.846200    -79.124410   
std                              NaN      0.051183      0.101086   

                      TFS_Alarm_Time               TFS_Arrival_Time  
count                          11214                          11214  
mean   2015-03-24 00:14:27.684501248  2015-03-24 00:19:27.843142656  
min              2011-01-01 05:49:17            2011-01-01 05:56:57  
25%       2013-02-11 19:32:11.500000     2013-02-11 19:37:10.500000  
50%              2015-03-03 11:13:33     2015-03-03 11:18:41.500000  
75%    2017-05-14 06:08:14.750000128            2017-05-14 06:12:44  
max              2019-06-30 22:44:55            2019-06-30 22:51:52  
std                              NaN                            NaN

##### The only thing perhaps out of the ordinary here is the max estimated_no_of_persons_displaced being 999

In [136]:
displaced_sorted = tfire.sort_values(by='Estimated_Number_Of_Persons_Displaced', ascending=False)

In [142]:
displaced_sorted.head(10)

Area_of_Origin  \
3866   Utility Shaft (eg. electrical wiring/phone, etc.)   
5323                                        Laundry Area   
10517  Sleeping Area or Bedroom (inc. patients room, ...   
1543                         Undetermined  (formerly 98)   
9329   HVAC Equipment Room (furnace room, water heate...   
7423                                        Laundry Area   
8699      Living Area (e.g. living, TV, recreation, etc)   
3970                                Sales, Showroom Area   
8083                             Cooking Area or Kitchen   
5731                             Cooking Area or Kitchen   

                           Business_Impact  Civilian_Casualties  \
3866                          Undetermined                    0   
5323       Not applicable (not a business)                    0   
10517                         Undetermined                    0   
1543       Not applicable (not a business)                    0   
9329              No business interruption                    0   
7423              No business interruption                    0   
8699              No business interruption                    0   
3970                          Undetermined                    0   
8083              No business interruption                    0   
5731   May resume operations within a week                    0   

       Count_of_Persons_Rescued  Estimated_Dollar_Loss  \
3866                          0                  500.0   
5323                          0                 5000.0   
10517                         0              1850000.0   
1543                          0               300000.0   
9329                          0                25000.0   
7423                          0                 1000.0   
8699                          0               200000.0   
3970                          0                20000.0   
8083                          0                50000.0   
5731                          0                 1000.0   

       Estimated_Number_Of_Persons_Displaced Ext_agent_app_or_defer_time  \
3866                                     999         2016-07-14 21:30:08   
5323                                     997         2016-06-25 21:37:45   
10517                                    997         2016-06-22 18:15:49   
1543                                     997         2018-12-24 04:05:32   
9329                                     997         2015-12-31 15:03:03   
7423                                     997         2013-05-10 06:17:00   
8699                                     997         2013-12-21 20:27:15   
3970                                     997         2014-09-02 16:09:31   
8083                                     997         2017-05-12 03:29:36   
5731                                     997         2015-06-11 09:59:51   

                                          Extent_Of_Fire  \
3866                        Confined to object of origin   
5323                        Confined to object of origin   
10517  Spread beyond building of origin, resulted in ...   
1543                                    Entire Structure   
9329            Spread beyond room of origin, same floor   
7423                        Confined to object of origin   
8699        Spread to other floors, confined to building   
3970             Confined to part of room/area of origin   
8083             Confined to part of room/area of origin   
5731             Confined to part of room/area of origin   

                  Fire_Alarm_System_Impact_on_Evacuation  \
3866   Not applicable: Occupant(s) first alerted by o...   
5323   Not applicable: No fire alarm system, no perso...   
10517  Not applicable: No fire alarm system, no perso...   
1543                                        Undetermined   
9329   Not applicable: No fire alarm system, no perso...   
7423   Not applicable: No fire alarm system, no perso...   
8699   Not applicable: Occupant(s) first alerted by o...   
3970   Not applicable: Occ

##### There are quite a few high values so I shall not perform any actions with this at the moment but there may be cause to action something later

### Export cleaned data

In [173]:
tfire.to_csv(r"C:\Users\agwsi\CF\6. Advanced Analytics and Dashboard design\Project folder\3. Prepared Data\Fire_Incidents_Cleaned.csv")